spec/unit/mutant/isolation/fork_spec.rb

```diff
@@ -0,0 +1,260 @@
+# The fork isolation is all about managing a series of systemcalls with proper error handling
+#
+# So creating a unit spec for this is challenging. Especially under mutation testing.
+# Hence we even have to implement our own message expectation mechanism, as rspec build in
+# expectations are not able to correctly specify a sequence of expectations where a specific
+# message is send twice.
+#
+# Also our replacement for rspec-expectations used here allows easier deduplication.
+RSpec.describe Mutant::Isolation::Fork do
+  let(:block_return)      { instance_double(Object, :block_return)      }
+  let(:block_return_blob) { instance_double(String, :block_return_blob) }
+  let(:devnull)           { instance_double(Proc, :devnull)             }
+  let(:io)                { class_double(IO)                            }
+  let(:isolated_block)    { -> { block_return }                         }
+  let(:marshal)           { class_double(Marshal)                       }
+  let(:process)           { class_double(Process)                       }
+  let(:pid)               { class_double(Fixnum)                        }
+  let(:reader)            { instance_double(IO, :reader)                }
+  let(:stderr)            { instance_double(IO, :stderr)                }
+  let(:stdout)            { instance_double(IO, :stdout)                }
+  let(:writer)            { instance_double(IO, :writer)                }
+  let(:nullio)            { instance_double(IO, :nullio)                }
+
+  describe '#call' do
+    let(:object) do
+      described_class.new(
+        devnull: devnull,
+        io:      io,
+        marshal: marshal,
+        process: process,
+        stderr:  stderr,
+        stdout:  stdout
+      )
+    end
+
+    subject { object.call(&isolated_block) }
+
+    class MessageExpectation
+      include Anima.new(:receiver, :selector, :arguments, :return, :yields, :exception)
+
+      # rubocop:disable ParameterLists
+      def self.new(receiver:, selector:, arguments: [], return: nil, yields: nil, exception: nil)
```

Comment:
Yeah, but my policy is to do it on the "scope that needs the least power". `.new` does not see a specific instance.